From cf7a5cba86fc2d3000c555b9568f7dd0f43bf0d4 Mon Sep 17 00:00:00 2001
Message-Id: <cf7a5cba86fc2d3000c555b9568f7dd0f43bf0d4.1650802266.git.geliang.tang@suse.com>
From: Jussi Maki <joamaki@gmail.com>
Date: Wed, 11 Aug 2021 12:36:27 +0000
Subject: [PATCH] selftests/bpf: Fix running of XDP bonding tests
Git-commit: cf7a5cba86fc2d3000c555b9568f7dd0f43bf0d4
Patch-mainline: v5.15-rc1
References: bsn#127
Modified-by-SEL: No


An "innocent" cleanup in the last version of the XDP bonding patchset moved
the "test__start_subtest" calls to the test main function, but I forgot to
reverse the condition, which lead to all tests being skipped. Fix it.

Fixes: 6aab1c81b98a ("selftests/bpf: Add tests for XDP bonding")
Signed-off-by: Jussi Maki <joamaki@gmail.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Link: https://lore.kernel.org/bpf/20210811123627.20223-1-joamaki@gmail.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/testing/selftests/bpf/prog_tests/xdp_bonding.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/testing/selftests/bpf/prog_tests/xdp_bonding.c b/tools/testing/selftests/bpf/prog_tests/xdp_bonding.c
index 6b186b4238d0..370d220288a6 100644
--- a/tools/testing/selftests/bpf/prog_tests/xdp_bonding.c
+++ b/tools/testing/selftests/bpf/prog_tests/xdp_bonding.c
@@ -493,20 +493,20 @@ void test_xdp_bonding(void)
 			   "xdp_redirect_multi_kern__open_and_load"))
 		goto out;
 
-	if (!test__start_subtest("xdp_bonding_attach"))
+	if (test__start_subtest("xdp_bonding_attach"))
 		test_xdp_bonding_attach(&skeletons);
 
 	for (i = 0; i < ARRAY_SIZE(bond_test_cases); i++) {
 		struct bond_test_case *test_case = &bond_test_cases[i];
 
-		if (!test__start_subtest(test_case->name))
+		if (test__start_subtest(test_case->name))
 			test_xdp_bonding_with_mode(
 				&skeletons,
 				test_case->mode,
 				test_case->xmit_policy);
 	}
 
-	if (!test__start_subtest("xdp_bonding_redirect_multi"))
+	if (test__start_subtest("xdp_bonding_redirect_multi"))
 		test_xdp_bonding_redirect_multi(&skeletons);
 
 out:
-- 
2.26.2

